### PR TITLE
Fix Felzenszwalb image segmentation

### LINKED ISSUE #2056: `min_size` is not strictly conformed in the implementation of felzenszwalb
## Description
With `min_size` specified, there're still some segments with sizes that less than it. I don't know if it is an inherent flaw of the algorithm.

## Way to reproduce

    >>> I = skimage.io.imread('dragonbaby.jpg')
    >>> fz = felzenszwalb(I, scale=300, sigma=0.8, min_size=80)
    >>> (fz==9).sum()
    1

dragonbaby

### PR DESCRIPTION
**Pull request by mljli**
## Description
Fixed a bug in post-processing and improved support for multi-channel images.

## References
Fixes #2056 and the issue discussed in #2134

### CODE REVIEW

File: skimage/segmentation/_felzenszwalb.py [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
-    For RGB images, the algorithm computes a separate segmentation for each
-    channel and then combines these. The combined segmentation is the
-    intersection of the separate segmentations on the color channels.
+    For multichannel images, the algorithm uses the euclidean distance between
+    pixels in channel space.
 
     Parameters
     ----------
-    image : (width, height, 3) or (width, height) ndarray
+    image : (height, width, N) or (height, width) ndarray
```
Comment (soupault) [inline skimage/segmentation/_felzenszwalb.py]: This should be:
`image : (M, N[, C]) array`.

File: skimage/segmentation/_felzenszwalb.py [hunk 2]
```diff
@@ -34,7 +33,7 @@ def felzenszwalb(image, scale=1, sigma=0.8, min_size=20):
 
     Returns
     -------
-    segment_mask : (width, height) ndarray
+    segment_mask : (height, width) ndarray
```
Comment (soupault) [inline skimage/segmentation/_felzenszwalb.py]: `segment_mask : (M, N) array`

File: skimage/segmentation/_felzenszwalb_cy.pyx [hunk 3]
```diff
@@ -12,48 +12,53 @@ from ..measure._ccomp cimport find_root, join_trees
...
+    Produces an oversegmentation of an image using a fast, minimum
+    spanning tree based clustering on the image grid.
     The number of produced segments as well as their size can only be
     controlled indirectly through ``scale``. Segment size within an image can
     vary greatly depending on local contrast.
 
     Parameters
     ----------
-    image: ndarray
+    image : (height, width, N) ndarray
```
Comment (soupault) [inline skimage/segmentation/_felzenszwalb_cy.pyx]: Same as above.

File: skimage/segmentation/_felzenszwalb_cy.pyx [hunk 4]
```diff
@@ -12,48 +12,53 @@ from ..measure._ccomp cimport find_root, join_trees
...
         Width of Gaussian smoothing kernel used in preprocessing.
         Larger sigma gives smother segment boundaries.
-    min_size: int, optional (default 20)
+    min_size : int, optional (default 20)
         Minimum component size. Enforced using postprocessing.
 
     Returns
     -------
-    segment_mask: (height, width) ndarray
+    segment_mask : (height, width) ndarray
```
Comment (soupault) [inline skimage/segmentation/_felzenszwalb_cy.pyx]: Same

File: skimage/segmentation/_felzenszwalb_cy.pyx [hunk 5]
```diff
@@ -12,48 +12,53 @@ from ..measure._ccomp cimport find_root, join_trees
...
         Integer mask indicating segment labels.
     """
-    if image.ndim != 2:
-        raise ValueError("This algorithm works only on single-channel 2d"
-                "images. Got image of shape %s" % str(image.shape))
+    if image.ndim != 3:
+        raise ValueError("This algorithm works only on 3d images. "
+                "Got image of shape %s" % str(image.shape))
```
Comment (soupault) [inline skimage/segmentation/_felzenszwalb_cy.pyx]: Fix indentation.

Comment (soupault) [inline skimage/segmentation/_felzenszwalb_cy.pyx]: And use `.format()` instead of `% ()`.

**Comment by soupault** [comment]
@mljli Thank you very much! Just a couple comments on style, otherwise :+1:.

**Comment by soupault** [comment]
@jmetz Are you OK with this? Or should @mljli build the PR on top of your commit?

**Comment by jmetz** [comment]
Hi @ahojnnes @soupault @mljli - apologies wasn't able to reply sooner; I guess being acknowledged for my contribution would be ideal... is it easy to add @mljli 's changes to this PR?

**Comment by mljli** [comment]
@soupault Done.

**Comment by mljli** [comment]
Hmm..Reopening this does not work as I expected. I'm creating a new one.

**Pull request closed without merging**